src/msg/msg_types.h

```diff
@@ -551,6 +551,27 @@ struct entity_addrvec_t {
       return out << av.v;
     }
   }
+
+  friend bool operator==(const entity_addrvec_t& l, const entity_addrvec_t& r) {
+    return l.v == r.v;
+  }
+  friend bool operator!=(const entity_addrvec_t& l, const entity_addrvec_t& r) {
+    return l.v != r.v;
+  }
+  friend bool operator<(const entity_addrvec_t& l, const entity_addrvec_t& r) {
+    for (unsigned p = 0; l.v.size() >= p && r.v.size() >= p; ++p) {
```

Comment:
could just use the `vector` 's overloaded operator defined in `<vector>`, it performs `lexicographical_compare()` under the hood. it's equivalent to what we have here.